### PR TITLE
WT-10954 Make convergence less likely for random values

### PR DESCRIPTION
**Pull request by pmacko86**

### CODE REVIEW

**Comment by pmacko86** [comment]
I added a new test, which I just called "random." It felt like a generic enough test that I didn't want to use the wtXXXX format for the name, but I can if that's preferred.

I used the "pull_request" tag for it for now, but I'll be happy to change it to something less frequent. I wasn't sure what is the right tag, so I'd appreciate a suggestion :) Thanks!

File: src/support/rand.c [hunk 1]
```diff
@@ -132,15 +150,22 @@ __wt_random(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visibility
     z = M_Z(rnd);
 
     /*
-     * Check if the value goes to 0 (from which we won't recover), and reset to the initial state.
+     * Check if either of the two value goes to 0 (from which we won't recover), and reset it to the
```
Comment (ddanderson) [inline src/support/rand.c:153]: two value -> two values

File: src/support/rand.c [hunk 2]
```diff
@@ -132,15 +150,22 @@ __wt_random(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visibility
...
      * This has additional benefits if a caller fails to initialize the state, or initializes with a
      * seed that results in a short period.
      */
-    if (z == 0 || w == 0) {
-        __wt_random_init(&rnd);
-        w = M_W(rnd);
-        z = M_Z(rnd);
-    }
+    if (w == 0)
+        M_W(rnd) = w = DEFAULT_SEED_W;
```
Comment (ddanderson) [inline src/support/rand.c:166]: The assignment to `M_W` is not needed here, only to `w` (and `M_Z` later).  Sometimes redundancy (or extra safety) is warranted, but here I think it works logically to have a single place that the "final" Z and W are adjusted.

File: src/support/rand.c [hunk 3]
```diff
@@ -132,15 +150,22 @@ __wt_random(WT_RAND_STATE volatile *rnd_state) WT_GCC_FUNC_ATTRIBUTE((visibility
...
+    if (z == 0)
+        M_Z(rnd) = z = DEFAULT_SEED_Z;
 
     M_Z(rnd) = z = 36969 * (z & 65535) + (z >> 16);
     M_W(rnd) = w = 18000 * (w & 65535) + (w >> 16);
```
Comment (ddanderson) [inline src/support/rand.c:171]: While you're in here, can you swap these two?  Everywhere else W comes before Z.

**Review by ddanderson — APPROVED** [review]
LGTM modulo some smaller things.  Thanks so much for creating the csuite program - just perfect!  I assume you've run that program on the previous version of `__wt_random`?

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

One suggestion about rewording a comment.

**Comment by pmacko86** [comment]
> LGTM modulo some smaller things. Thanks so much for creating the csuite program - just perfect! I assume you've run that program on the previous version of `__wt_random`?

Yes, I ran it first with the original `__wt_random`.

**Pull request merged**